packages/common/package.json

```diff
@@ -6,7 +6,7 @@
   "main": "src/index.ts",
   "scripts": {
     "lint": "yarn lint:run --ext .ts,.tsx src rollup-configs.js",
-    "test": "echo 'Write some tests!'"
+    "test": "yarn --cwd ../.. test packages/common"
```

Comment:
This is the Console way of running tests, with all common dependencies crammed into root `package.json` file. This is not what Yarn 2+ recommends.

I'd change this to `jest .` or similar.

Check the `lint` script above, it follows the same pattern. Root package has `"lint": "yarn workspaces foreach run lint"` and specific packages run ESLint on files/directories within that package.

Reply:
Unfortunately that won't work `jest` would have to be installed in each package and the config would have to be shared (that's OK-ish), however collecting codecov would be nightmare (each package would have its own coverage report). I've tried this in a couple of monorepos and unless there's been a big change in jest this is the way.

There's an option to have package just for the jest config and run it similarly how `lint:run` is done. I could do that, but that brings extra complexity to the play. I'd rather go with the standard "One jest in root and run it from there". 

Should we want to have more granular config (one for react and the other one for eslint for instance) we can use the `projects` field https://github.com/facebook/jest/issues/5420#issuecomment-369714348.

Reply:
If you find something let me know, I've been looking for the non-root approach of jest testig for quite some time and couldn't find anything. There's the `packages` config in jest we could use, but that still didn't allow running jest from within each package. After all the jest is installed only in root, we'd have to install it to every package and that is hell to maintain.

Reply:
> `jest` would have to be installed in each package

This is the standard approach for monorepos using Yarn 2+

Reply:
Each package is a standalone module, what's conceptually wrong with each one having its own code coverage report?

Reply:
If we'd were to add jest as dev dependency of each package that would be nightmare to keep up to date (been there, done that). Codecov monitors cpde coverage per repository and not per package. After all it tracks a coverage ot code.

We would still have to add config for each package, again extra work on our side to keep everything in order.

And the last thing is that jest takes a couple seconds to start for each instance.

As I said, I've tried having jest configs per package and it didn't work. If you want we can give it another try, but I would mark that as Spike and not stall this PR anymore